dvc/command/stage.py

```diff
@@ -254,6 +276,12 @@ def _add_common_args(parser):
             "This doesn't affect any DVC operations."
         ),
     )
+    parser.add_argument(
+        "cmd",
+        nargs=argparse.REMAINDER,
```

Comment:
Looks like this is greedy.

Reply:
We might need to move this back to `-n` like in `dvc run`, if I could not find a way.

Reply:
`-n` is a safe choice

Reply:
Okay, so it is greedy. The following works:
```console
$ dvc stage add -d foo -o bar stage_name echo foo bar
```
But, the following does not:
```console
$ dvc stage add stage_name -d foo -o bar echo foo bar
```
It seems `REMAINDER` starts eating up rest of the arguments when the parser starts parsing positional arguments.

Reply:
Yup, looks like it's correct. Note that `REMAINDER` has been removed from documentation as it's considered buggy.
https://bugs.python.org/issue17050

Reply:
Though it seems, we can achieve best of the both worlds (remainder + `-c`) by making `-c` a remainder rather than a positional argument.

Reply:
My vote is to mimic the 1.0 dvc run syntax unless there's a compelling justification for changing it:

`$ dvc stage add -n stage_name -d foo -o bar rest of the things are command`

Any concerns besides what was in #5342?

Reply:
Moving back to `-n` then.